std/assembly/internal/date.ts

```diff
@@ -0,0 +1,381 @@
+        728659, // 1996
+        729025, // 1997
+        729390, // 1998
+        729755, // 1999
+        730120, // 2000
+        730486, // 2001
+        730851, // 2002
+        731216, // 2003
+        731581, // 2004
+        731947, // 2005
+        732312, // 2006
+        732677, // 2007
+        733042, // 2008
+        733408, // 2009
+        733773, // 2010
+        734138, // 2011
+        734503, // 2012
+        734869, // 2013
+        735234, // 2014
+        735599, // 2015
+        735964, // 2016
+        736330, // 2017
+        736695, // 2018
+        737060, // 2019
+        737425, // 2020
+        737791, // 2021
+        738156, // 2022
+        738521, // 2023
+        738886, // 2024
+        739252, // 2025
+        739617, // 2026
+        739982, // 2027
+        740347, // 2028
+        740713, // 2029
+        741078, // 2030
+        741443, // 2031
+        741808, // 2032
+        742174, // 2033
+        742539, // 2034
+        742904, // 2035
+        743269, // 2036
+        743635, // 2037
+        744000, // 2038
+        744365 // 2039
+    ];
+
+    cdate: CalendarDate;
+
+    isNormalized(): bool {
+      return this.cdate.isNormalized;
+    }
+
+    /**
+     * Returns the Geogorian year number of the given fixed date.
+     */
+    private getYearFromFixedDate(fixedDate: i32): i32 {
+        var d0: i32;
+        var d1: i32, d2: i32, d3: i32;
+        var n400: i32, n100: i32, n4: i32, n1: i32;
+        var year: i32;
+        if ( fixedDate > 0) {
+          d0 = fixedDate - 1;
+          n400 = d0 / 146097;
+          d1 = d0 % 146097;
+          n100 = d1 / 36524;
+          d2 = d1 % 36524;
+          n4 = d2 / 1461;
+          d3 = d2 % 1461;
+          n1 = d3 / 365;
+          // d4 = (d3 % 365) + 1;
+        } else {
+          d0 = fixedDate - 1;
+          n400 = CalendarUtils.floorDivide(d0, 146097);
+          d1 = CalendarUtils.mod(d0, 146097);
+          n100 = CalendarUtils.floorDivide(d1, 36524);
+          d2 = CalendarUtils.mod(d1, 36524);
+          n4 = CalendarUtils.floorDivide(d2, 1461);
+          d3 = CalendarUtils.mod(d2, 1461);
+          n1 = CalendarUtils.floorDivide(d3, 365);
+          // d4 = CalendarUtils.mod(d3, 365) + 1;
+        }
+        year = 400 * n400 + 100 * n100 + 4 * n4 + n1;
+        if (!(n100 == 4 || n1 == 4)) {
+          ++year;
+        }
+        return year;
+      }
+
+      /**
+       * Returns the days of the specified date.
+       * @param year the year of the date
+       * @param month the month of the date, 1-based
+       * @param dayOfMonth the day of the month, 1-based
+       */
+      private getFixedDate(year: i32, month: i32, dayOfMonth: i32): i32 {
+
+        var prevyear = year - 1;
+        var days = dayOfMonth;
+
+        if (prevyear >= 0) {
+          days += (365 * prevyear)
+          + (prevyear / 4)
+          - (prevyear / 100)
+          + (prevyear / 400)
+          + ((367 * month - 362) / 12);
```

Comment:
Hmm, btw `moment` it seems calculate this as `365 * year + countLeaps(year)`:
http://jsfiddle.net/qb3d9pzs